precompiles/xtokens/src/lib.rs

```diff
@@ -270,15 +281,18 @@ where
 	) -> EvmResult<PrecompileOutput> {
 		let mut input = handle.read_input()?;
 		input.expect_arguments(4)?;
-		let non_mapped_currencies: Vec<Currency> = input.read::<Vec<Currency>>()?;
-		let max_assets = MaxAssetsForTransfer::<Runtime>::get();
+		let non_mapped_currencies: BoundedVec<Currency, GetMaxAssets<Runtime>> = input.read()?;
+		let non_mapped_currencies = non_mapped_currencies.into_vec();
+
+		// Now checked using BoundedVec.
+		// TODO: Improve error message customization.
```

Comment:
Is this TODO still relevant?

Reply:
Currently it will return a general revert message. I have some ideas to improve the crate to make better error messages but that will go into another PR. I'll remove this comment for now.